File: c-tokens/source/number_token.c
#include "number_token.h"

// Number Tokens
const NumberToken ZERO = '0';
const NumberToken ONE = '1';
const NumberToken TWO = '2';
const NumberToken THREE = '3';
const NumberToken FOUR = '4';
const NumberToken FIVE = '5';
const NumberToken SIX = '6';
const NumberToken SEVEN = '7';
const NumberToken EIGHT = '8';
const NumberToken NINE = '9';

// Number Token Vector
const NumberToken numbers_vector()  {
    const NumberToken numbers[10] = {
        ZERO,
        ONE,
        TWO,
        THREE,
        FOUR,
        FIVE,
        SIX,
        SEVEN,
        EIGHT,
        NINE,
    };

    return *numbers;
}

File: tokens/src/number_token.rs
use std::{
    primitive::char,
    vec::Vec,
};

// Number Tokens
pub const ZERO: char = '0';
pub const ONE: char = '1';
pub const TWO: char = '2';
pub const THREE: char = '3';
pub const FOUR: char = '4';
pub const FIVE: char = '5';
pub const SIX: char = '6';
pub const SEVEN: char = '7';
pub const EIGHT: char = '8';
pub const NINE: char = '9';

// Number Token Vector
pub fn numbers_vector() -> Vec<char> {
    let numbers: Vec<char> = Vec::from([
        ZERO,
        ONE,
        TWO,
        THREE,
        FOUR,
        FIVE,
        SIX,
        SEVEN,
        EIGHT,
        NINE,
    ]);

    return numbers;
}
